documentation/general/dotnet-run-file.md

```diff
@@ -172,56 +172,54 @@ App/Program2/bin/
 App/Program2/obj/
 ```
 
-## Package references
+## Directives for project metadata
 
-It is possible to specify NuGet package references via the `#package` directive.
+It is possible to specify some project metadata via [ignored C# directives][ignored-directives].
+Directives `sdk`, `package`, and `property` are translated into `<Project Sdk="...">`, `<PackageReference>`, and `<Property>` project elements, respectively.
+Other directives result in a warning, reserving them for future use.
 
 ```cs
-#package Newtonsoft.Json 13.0.1
+#:sdk      Microsoft.NET.Sdk.Web
+#:property TargetFramework=net11.0
+#:property LangVersion=preview
+#:package  System.CommandLine=2.0.0-*
 ```
 
-The C# language needs to be updated to ignore these directives (instead of failing the compilation).
-See [the corresponding language proposal][pound].
+The value must be separated from the name of the directive by white space and any leading and trailing white space is not considered part of the value.
+The value of `#:sdk` is injected into `<Project Sdk="{0}">` as is.
```

Comment:
I thought you had the SDK not actually in the SDK= part but imported manually so you could overwrite certain bits? Is that changing?

Reply:
No, that's not changing, but I would consider that to be an internal implementation detail (and it should go away if/when NuGet fixes their targets to not fail on virtual project files). Having `<Project Sdk="..."> ... </Project>` should be equivalent to `<Project> <Import props /> ... <Import targets /> </Project>` (I believe the former is just "syntactic sugar" for the latter, but I might be wrong).

Reply:
Fair enough